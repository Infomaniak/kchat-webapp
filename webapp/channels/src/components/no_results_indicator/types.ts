// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export enum NoResultsVariant {
    Search = 'Search',
    Files = 'Files',
    ChannelSearch = 'ChannelSearch',
    Mentions = 'Mentions',
    FlaggedPosts = 'FlaggedPosts',
    PinnedPosts = 'PinnedPosts',
    ChannelFiles = 'ChannelFiles',
    ChannelFilesFiltered = 'ChannelFilesFiltered',
    UserGroups = 'UserGroups',
    UserGroupMembers = 'UserGroupMembers',
    UserGroupsArchived = 'UserGroupsArchived',
}

export enum NoResultsLayout {
    Vertical,
    Horizontal,
}
