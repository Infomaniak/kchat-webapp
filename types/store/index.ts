// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {KSuiteBridge} from '@infomaniak/ksuite-bridge';

import {GlobalState as BaseGlobalState} from '@mattermost/types/store';

import {PluginsState} from './plugins';
import {ViewsState} from './views';

export type DraggingState = {
    state?: string;
    type?: string;
    id?: string;
}

export type GlobalState = BaseGlobalState & {
    plugins: PluginsState;
    ksuite_bridge: {
        bridge: KSuiteBridge;
    };
    storage: {
        storage: Record<string, any>;
        initialized: boolean;
    };
    views: ViewsState;
};
