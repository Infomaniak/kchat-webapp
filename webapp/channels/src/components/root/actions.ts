// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {History} from 'history';

import type {UserProfile} from '@mattermost/types/users';

import {getFirstAdminSetupComplete} from 'mattermost-redux/actions/general';
import {getProfiles} from 'mattermost-redux/actions/users';
import {General} from 'mattermost-redux/constants';
import {getIsOnboardingFlowEnabled} from 'mattermost-redux/selectors/entities/preferences';
import {getActiveTeamsList} from 'mattermost-redux/selectors/entities/teams';
import {checkIsFirstAdmin, getCurrentUser, isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
import type {ThunkActionFunc} from 'mattermost-redux/types/actions';

import * as GlobalActions from 'actions/global_actions';

import {StoragePrefixes} from 'utils/constants';

export function redirectToOnboardingOrDefaultTeam(history: History): ThunkActionFunc<void> {
    return async (dispatch, getState) => {
        const state = getState();
        const isUserAdmin = isCurrentUserSystemAdmin(state);
        if (!isUserAdmin) {
            GlobalActions.redirectUserToDefaultTeam();
            return;
        }

        const teams = getActiveTeamsList(state);

        const onboardingFlowEnabled = getIsOnboardingFlowEnabled(state);

        if (teams.length > 0 || !onboardingFlowEnabled) {
            GlobalActions.redirectUserToDefaultTeam();
            return;
        }

        const firstAdminSetupComplete = await dispatch(getFirstAdminSetupComplete());
        if (firstAdminSetupComplete?.data) {
            GlobalActions.redirectUserToDefaultTeam();
            return;
        }

        const profilesResult = await dispatch(getProfiles(0, General.PROFILE_CHUNK_SIZE, {roles: General.SYSTEM_ADMIN_ROLE}));
        if (profilesResult.error) {
            GlobalActions.redirectUserToDefaultTeam();
            return;
        }
        const currentUser = getCurrentUser(getState());
        const adminProfiles = profilesResult.data?.reduce(
            (acc: Record<string, UserProfile>, curr: UserProfile) => {
                acc[curr.id] = curr;
                return acc;
            },
            {},
        );
        if (adminProfiles && checkIsFirstAdmin(currentUser, adminProfiles)) {
            history.push('/preparing-workspace');
            return;
        }

        GlobalActions.redirectUserToDefaultTeam();
    };
}

export function handleLoginLogoutSignal(e: StorageEvent): ThunkActionFunc<void> {
    return (dispatch, getState) => {
        // when one tab on a browser logs out, it sets __logout__ in localStorage to trigger other tabs to log out
        const isNewLocalStorageEvent = (event: StorageEvent) => event.storageArea === localStorage && event.newValue;

        if (e.key === StoragePrefixes.LOGOUT && isNewLocalStorageEvent(e)) {
            console.log('detected logout from a different tab'); //eslint-disable-line no-console
            GlobalActions.emitUserLoggedOutEvent('/', false, false);
        }
        if (e.key === StoragePrefixes.LOGIN && isNewLocalStorageEvent(e)) {
            const isLoggedIn = getCurrentUser(getState());

            // make sure this is not the same tab which sent login signal
            // because another tabs will also send login signal after reloading
            if (isLoggedIn) {
                return;
            }

            // detected login from a different tab
            function reloadOnFocus() {
                location.reload();
            }
            window.addEventListener('focus', reloadOnFocus);
        }
    };
}
