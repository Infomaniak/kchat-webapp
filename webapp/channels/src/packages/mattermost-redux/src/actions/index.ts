// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import * as admin from './admin';
import * as bots from './bots';
import * as channels from './channels';
import * as cloud from './cloud';
import * as emojis from './emojis';
import * as errors from './errors';
import * as files from './files';
import * as general from './general';
import * as gifs from './gifs';
import * as groups from './groups';
import * as helpers from './helpers';
import * as insights from './insights';
import * as integrations from './integrations';
import * as jobs from './jobs';
import * as posts from './posts';
import * as preferences from './preferences';
import * as roles from './roles';
import * as schemes from './schemes';
import * as search from './search';
import * as teams from './teams';
import * as timezone from './timezone';
import * as users from './users';
import * as websocket from './websocket';

export {
    admin,
    bots,
    channels,
    cloud,
    errors,
    emojis,
    files,
    general,
    groups,
    gifs,
    integrations,
    helpers,
    jobs,
    posts,
    preferences,
    roles,
    schemes,
    search,
    teams,
    timezone,
    websocket,
    users,
    insights,
};

