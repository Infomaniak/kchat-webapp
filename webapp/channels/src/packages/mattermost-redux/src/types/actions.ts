// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {Action, AnyAction, Dispatch} from 'redux';
import type {ThunkAction} from 'redux-thunk';

import type {GlobalState} from '@mattermost/types/store';

/**
 * This file extends Redux's Dispatch type and bindActionCreators function to support Thunk actions by default.
 *
 * It specifically requires those action creators to return ThunkAction-derived types.
 */
import 'redux-thunk/extend-redux';

export type DispatchFunc = Dispatch;
export type GetStateFunc = () => GlobalState;

/**
 * ActionResult should be the return value of most Thunk action creators.
 */
export type ActionResult<Data = any, Error = any> = {
    data?: Data;
    error?: Error;
};

/**
 * ActionFunc should be the return type of most non-async Thunk action creators. If that action requires web app state,
 * the second type parameter should be used to pass the version of GlobalState from 'types/store'.
 */
export type ActionFunc<Data = unknown, State extends GlobalState = GlobalState> = ThunkAction<ActionResult<Data>, State, unknown, Action>;

/**
 * ActionFuncAsync should be the return type of most async Thunk action creators. If that action requires web app state,
 * the second type parameter should be used to pass the version of GlobalState from 'types/store'.
 */
export type ActionFuncAsync<Data = unknown, State extends GlobalState = GlobalState> = ThunkAction<Promise<ActionResult<Data>>, State, unknown, Action>;

/**
 * ThunkActionFunc is a type that extends ActionFunc with defaults that match our other ActionFunc variants to save
 * users from having to manually specify GlobalState and other arguments.
 *
 * ActionFunc or ActionFuncAsync should generally be preferred, but this type is available for legacy code.
 */
export type ThunkActionFunc<ReturnType, State extends GlobalState = GlobalState> = ThunkAction<ReturnType, State, unknown, AnyAction>;
