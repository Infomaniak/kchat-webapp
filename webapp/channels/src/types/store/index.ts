// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {KSuiteBridge} from '@infomaniak/ksuite-bridge';

import type {GlobalState as BaseGlobalState} from '@mattermost/types/store';

import type {PluginsState} from './plugins';
import type {ViewsState} from './views';

export type DraggingState = {
    state?: string;
    type?: string;
    id?: string;
}

export type GlobalState = BaseGlobalState & {
    plugins: PluginsState;
    ksuite_bridge: KSuiteBridge;
    kdrive: {
        toast: {
            message: string;
        };
    };
    storage: {
        storage: Record<string, any>;
        initialized: boolean;
    };
    views: ViewsState;
};
