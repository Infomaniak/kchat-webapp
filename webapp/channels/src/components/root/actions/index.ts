// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {History} from 'history';

import type {ServerError} from '@mattermost/types/errors';
import type {UserProfile} from '@mattermost/types/users';

import {GeneralTypes} from 'mattermost-redux/action_types';
import {logError} from 'mattermost-redux/actions/errors';
import {getClientConfig, getLicenseConfig, getFirstAdminSetupComplete} from 'mattermost-redux/actions/general';
import {redirectToErrorPageIfNecessary} from 'mattermost-redux/actions/helpers';
import {getMyPreferences} from 'mattermost-redux/actions/preferences';
import {getMyKSuites, getMyTeamMembers, getMyTeamUnreads} from 'mattermost-redux/actions/teams';
import {getMe, getProfiles} from 'mattermost-redux/actions/users';
import {Client4} from 'mattermost-redux/client';
import {General} from 'mattermost-redux/constants';
import {isCollapsedThreadsEnabled, getIsOnboardingFlowEnabled, getTeamsOrderPreference} from 'mattermost-redux/selectors/entities/preferences';
import {getActiveTeamsList, getTeams} from 'mattermost-redux/selectors/entities/teams';
import {checkIsFirstAdmin, getCurrentUser, isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
import {getLastKSuiteSeenId} from 'mattermost-redux/utils/team_utils';

import {getMyMeets} from 'actions/calls';
import {redirectUserToDefaultTeam, emitUserLoggedOutEvent} from 'actions/global_actions';
import {updateTeamsOrderForUser} from 'actions/team_actions';

import {getHistory} from 'utils/browser_history';
import {ActionTypes, StoragePrefixes} from 'utils/constants';
import {isDesktopApp, isMacApp, isNotMacMas} from 'utils/user_agent';

import type {ActionFuncAsync, ThunkActionFunc} from 'types/store';
import type {Translations} from 'types/store/i18n';

export type TranslationPluginFunction = (locale: string) => Translations

/**
 * This function meant to be used in root.tsx component loads config, license and if user is logged in, it loads user and its related data.
 */
export function loadConfigAndMe(): ThunkActionFunc<Promise<{isLoaded: boolean; isMeRequested?: boolean}>> {
    return async (dispatch, getState) => {
        // attempt to load config and license regardless if user is logged in or not
        try {
            await Promise.all([
                dispatch(getClientConfig()),
                dispatch(getLicenseConfig()),
            ]);

            let redirect = true;
            if (!isMacApp() || !isNotMacMas()) {
                redirect = false;
            }

            if (redirect) {
                const forceMigrationError: ServerError = {
                    message: 'Maintenance mode',
                    status_code: 1,
                };
                redirectToErrorPageIfNecessary(forceMigrationError);
            }
        } catch (error) {
            dispatch(logError(error as ServerError));
            return {
                isLoaded: false,
            };
        }

        // Load user and its related data now that we know that user is logged in
        const serverVersion = getState().entities.general.serverVersion || Client4.getServerVersion();
        dispatch({type: GeneralTypes.RECEIVED_SERVER_VERSION, data: serverVersion});
        const kSuiteCall = await dispatch(getMyKSuites());
        const kSuites = getTeams(getState());

        const suiteArr = Object.values(kSuites);

        if (suiteArr.length > 0) {
            const lastKSuiteSeenId = getLastKSuiteSeenId();
            const sortedSuites = suiteArr.sort((a, b) => {
                if (a.id === lastKSuiteSeenId) {
                    return -1;
                }
                if (b.id === lastKSuiteSeenId) {
                    return 1;
                }
                return b.update_at - a.update_at;
            });
            const lastKSuiteSeen = sortedSuites[0];

            if (isDesktopApp()) {
                window.postMessage({
                    type: 'switch-server',
                    data: lastKSuiteSeen.display_name,
                }, window.origin);
            }

            try {
                await Promise.all([
                    dispatch(getMe()),
                    dispatch(getMyPreferences()),
                    dispatch(getMyTeamMembers()),
                    dispatch(getMyMeets()),
                ]);
                dispatch(getMyTeamUnreads(isCollapsedThreadsEnabled(getState())));

                if (kSuiteCall?.data && Array.isArray(kSuiteCall.data)) {
                    // IK change: The desktop logic for team order is based on the 'teams_order' preference.
                    // Therefore, we need to update the preference while respecting the order already established by the desktop.
                    // If there is a new team, we add it to the end of the array.
                    // If one of the teams in the preference has been deleted, we remove it.

                    const fetchedTeamIds = kSuiteCall.data.map((team) => team.id);
                    const state = getState();

                    // IK change: getTeamsOrderPreference returns an object, not an array
                    // We need to convert the CSV string in prefObj.value into a proper array
                    const prefObj = getTeamsOrderPreference(state);
                    const currentPreferences: string[] = prefObj?.value ? prefObj.value.split(',') : [];

                    // Filter out any preferences that no longer exist in fetchedTeamIds
                    const cleanedPreferences = currentPreferences.filter((id) => fetchedTeamIds.includes(id));

                    // Find any new teams that aren't already in the preferences
                    const missingTeamIds = fetchedTeamIds.filter((id) => !cleanedPreferences.includes(id));

                    // Combine the cleaned preferences with the missing teams appended at the end
                    const updatedPreferences = [...cleanedPreferences, ...missingTeamIds];

                    // Ensure uniqueness just in case
                    const uniquePreferences = [...new Set(updatedPreferences)];

                    // Check if the preferences actually changed
                    const hasChanged =
        uniquePreferences.length !== currentPreferences.length ||
        uniquePreferences.some((id, index) => id !== currentPreferences[index]);

                    // If there was a change, update the user's team order preferences
                    if (hasChanged) {
                        // eslint-disable-next-line no-console
                        console.log('[loadConfigAndMe] Syncing team order preferences after fetch:', uniquePreferences);
                        updateTeamsOrderForUser(uniquePreferences);
                    }
                }
            } catch (error) {
                dispatch(logError(error as ServerError));
                return {
                    isLoaded: false,
                };
            }
        } else if (!isDesktopApp()) {
            // we should not use getHistory in mattermost-redux since it is an import from outside the package, but what else can we do
            if (kSuiteCall && kSuiteCall.data) {
                getHistory().push('/error?type=no_ksuite');
            }
        }

        return {
            isLoaded: true,
            isMeRequested: true,
        };
    };
}

export function registerCustomPostRenderer(type: string, component: any, id: string): ActionFuncAsync {
    return async (dispatch) => {
        // piggyback on plugins state to register a custom post renderer
        dispatch({
            type: ActionTypes.RECEIVED_PLUGIN_POST_COMPONENT,
            data: {
                postTypeId: id,
                pluginId: id,
                type,
                component,
            },
        });
        return {data: true};
    };
}

export function redirectToOnboardingOrDefaultTeam(history: History, searchParams?: URLSearchParams): ThunkActionFunc<void> {
    return async (dispatch, getState) => {
        const state = getState();
        const isUserAdmin = isCurrentUserSystemAdmin(state);
        if (!isUserAdmin) {
            redirectUserToDefaultTeam(searchParams);
            return;
        }

        const teams = getActiveTeamsList(state);

        const onboardingFlowEnabled = getIsOnboardingFlowEnabled(state);

        if (teams.length > 0 || !onboardingFlowEnabled) {
            redirectUserToDefaultTeam(searchParams);
            return;
        }

        const firstAdminSetupComplete = await dispatch(getFirstAdminSetupComplete());
        if (firstAdminSetupComplete?.data) {
            redirectUserToDefaultTeam(searchParams);
            return;
        }

        const profilesResult = await dispatch(getProfiles(0, General.PROFILE_CHUNK_SIZE, {roles: General.SYSTEM_ADMIN_ROLE}));
        if (profilesResult.error) {
            redirectUserToDefaultTeam(searchParams);
            return;
        }
        const currentUser = getCurrentUser(getState());
        const adminProfiles = profilesResult.data?.reduce(
            (acc: Record<string, UserProfile>, curr: UserProfile) => {
                acc[curr.id] = curr;
                return acc;
            },
            {},
        );
        if (adminProfiles && checkIsFirstAdmin(currentUser, adminProfiles)) {
            history.push('/preparing-workspace');
            return;
        }

        redirectUserToDefaultTeam(searchParams);
    };
}

export function handleLoginLogoutSignal(e: StorageEvent): ThunkActionFunc<void> {
    return (dispatch, getState) => {
        // when one tab on a browser logs out, it sets __logout__ in localStorage to trigger other tabs to log out
        const isNewLocalStorageEvent = (event: StorageEvent) => event.storageArea === localStorage && event.newValue;

        if (e.key === StoragePrefixes.LOGOUT && isNewLocalStorageEvent(e)) {
            console.log('detected logout from a different tab'); //eslint-disable-line no-console
            emitUserLoggedOutEvent('/', false, false);
        }
        if (e.key === StoragePrefixes.LOGIN && isNewLocalStorageEvent(e)) {
            const isLoggedIn = getCurrentUser(getState());

            // make sure this is not the same tab which sent login signal
            // because another tabs will also send login signal after reloading
            if (isLoggedIn) {
                return;
            }

            // detected login from a different tab
            function reloadOnFocus() {
                location.reload();
            }
            window.addEventListener('focus', reloadOnFocus);
        }
    };
}
