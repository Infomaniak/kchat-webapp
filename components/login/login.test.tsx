// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow, mount} from 'enzyme';
import {FormattedMessage, IntlProvider} from 'react-intl';

import AlertBanner from 'components/alert_banner';
import ExternalLoginButton from 'components/external_login_button/external_login_button';
import LoadingScreen from 'components/loading_screen';
import Login from 'components/login/login';
import Input from 'components/widgets/inputs/input/input';
import PasswordInput from 'components/widgets/inputs/password_input/password_input';
import SaveButton from 'components/save_button';

import users from 'mattermost-redux/actions/users';
import {RequestStatus} from 'mattermost-redux/constants';
import {ActionFunc} from 'mattermost-redux/types/actions';
import {ClientConfig} from '@mattermost/types/config';
import LocalStorageStore from 'stores/local_storage_store';
import {GlobalState} from 'types/store';
import Constants, {WindowSizes} from 'utils/constants';

let mockState: GlobalState;
let mockLocation = {pathname: '', search: '', hash: ''};
const mockHistoryReplace = jest.fn();
const mockHistoryPush = jest.fn();
const mockLicense = {IsLicensed: 'false'};
let mockConfig: Partial<ClientConfig>;

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux') as typeof import('react-redux'),
    useSelector: (selector: (state: typeof mockState) => unknown) => selector(mockState),
    useDispatch: jest.fn(() => (action: ActionFunc) => action),
}));

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom') as typeof import('react-router-dom'),
    useLocation: () => mockLocation,
    useHistory: () => ({
        replace: mockHistoryReplace,
        push: mockHistoryPush,
    }),
}));

jest.mock('mattermost-redux/selectors/entities/general', () => ({
    ...jest.requireActual('mattermost-redux/selectors/entities/general') as typeof import('mattermost-redux/selectors/entities/general'),
    getLicense: () => mockLicense,
    getConfig: () => mockConfig,
}));

describe('components/login/Login', () => {
    beforeEach(() => {
        mockLocation = {pathname: '', search: '', hash: ''};

        LocalStorageStore.setWasLoggedIn(false);

        mockState = {
            entities: {
                general: {
                    config: {},
                    license: {},
                },
                users: {
                    currentUserId: '',
                    profiles: {
                        user1: {
                            id: 'user1',
                            roles: '',
                        },
                    },
                },
                teams: {
                    currentTeamId: 'team1',
                    teams: {
                        team1: {
                            id: 'team1',
                            name: 'team-1',
                            displayName: 'Team 1',
                        },
                    },
                    myMembers: {
                        team1: {roles: 'team_role'},
                    },
                },
            },
            requests: {
                users: {
                    logout: {
                        status: RequestStatus.NOT_STARTED,
                    },
                },
            },
            storage: {
                initialized: true,
            },
            views: {
                browser: {
                    windowSize: WindowSizes.DESKTOP_VIEW,
                },
            },
        } as unknown as GlobalState;

        mockConfig = {
            EnableLdap: 'false',
            EnableSaml: 'false',
            EnableSignInWithEmail: 'false',
            EnableSignInWithUsername: 'false',
            EnableSignUpWithEmail: 'false',
            EnableSignUpWithGitLab: 'false',
            EnableSignUpWithOffice365: 'false',
            EnableSignUpWithGoogle: 'false',
            EnableSignUpWithOpenId: 'false',
            EnableOpenServer: 'false',
            LdapLoginFieldName: '',
            GitLabButtonText: '',
            GitLabButtonColor: '',
            OpenIdButtonText: '',
            OpenIdButtonColor: '',
            SamlLoginButtonText: '',
            EnableCustomBrand: 'false',
            CustomBrandText: '',
            CustomDescriptionText: '',
            SiteName: 'Mattermost',
            ExperimentalPrimaryTeam: '',
        };
    });

    it('should match snapshot', () => {
        const wrapper = shallow(
            <Login/>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should match snapshot with base login', () => {
        mockConfig.EnableSignInWithEmail = 'true';

        const wrapper = shallow(
            <Login/>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should handle initializing when logout status success', () => {
        mockState.requests.users.logout.status = RequestStatus.SUCCESS;

        const intlProviderProps = {
            defaultLocale: 'en',
            locale: 'en',
            messages: {},
        };

        const wrapper = mount(
            <IntlProvider {...intlProviderProps}>
                <Login/>
            </IntlProvider>,
        );

        const loadingScreen = wrapper.find(LoadingScreen).first();
        expect(loadingScreen.find(FormattedMessage).first().props().defaultMessage).toEqual('Loading');
    });

    it('should handle initializing when storage not initalized', () => {
        mockState.storage.initialized = false;

        const intlProviderProps = {
            defaultLocale: 'en',
            locale: 'en',
            messages: {},
        };

        const wrapper = mount(
            <IntlProvider {...intlProviderProps}>
                <Login/>
            </IntlProvider>,
        );

        const loadingScreen = wrapper.find(LoadingScreen).first();
        expect(loadingScreen.find(FormattedMessage).first().props().defaultMessage).toEqual('Loading');
    });
});
