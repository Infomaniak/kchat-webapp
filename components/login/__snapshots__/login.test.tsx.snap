// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/login/Login should match snapshot 1`] = `
<div
  className="login-body"
>
  <div
    className="login-body-content"
  >
    <LoadingIk
      position="relative"
      style={Object {}}
    />
  </div>
</div>
`;

exports[`components/login/Login should match snapshot with base login 1`] = `
<div
  className="login-body"
>
  <div
    className="login-body-content"
  >
    <LoadingIk
      position="relative"
      style={Object {}}
    />
  </div>
</div>
`;
